x/evm/keeper/msg_server.go

```diff
@@ -119,6 +120,48 @@ func (s msgServer) ConfirmGatewayTx(c context.Context, req *types.ConfirmGateway
 
 	return &types.ConfirmGatewayTxResponse{}, nil
 }
+func (s msgServer) ConfirmGatewayTxs(c context.Context, req *types.ConfirmGatewayTxsRequest) (*types.ConfirmGatewayTxsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	chain, ok := s.nexus.GetChain(ctx, req.Chain)
+	if !ok {
+		return nil, fmt.Errorf("%s is not a registered chain", req.Chain)
+	}
+
+	if err := validateChainActivated(ctx, s.nexus, chain); err != nil {
+		return nil, err
+	}
+
+	keeper, err := s.ForChain(ctx, chain.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	gatewayAddress, ok := keeper.GetGatewayAddress(ctx)
+	if !ok {
+		return nil, fmt.Errorf("axelar gateway address not set")
+	}
+
+	snapshot, err := s.createSnapshot(ctx, chain)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
why is this extracted out here?

Reply:
I took participants out, extracted out to get snapshot participants addresses

before we have 
```
message ConfirmGatewayTxStarted {
  bytes tx_id = 1;
  ...
  vote.exported.v1beta1.PollParticipants participants = 5;
}
```
where PollParticipants is 
```
message PollParticipants {
  uint64 poll_id = 1;
  repeated bytes participants = 2;
}
```
now changed to
```
message ConfirmGatewayTxsStarted {
  repeated PollMapping poll_mappings = 1;
  ...
  repeated bytes participants = 5;
}
```

Reply:
oh I see, so it's necessary for the event. In that case I suggest making `CreateSnapshot` public to signify that it's ok to call it from outside the keeper